duration.go

```diff
@@ -22,28 +26,48 @@ func (dur Duration) MarshalText() ([]byte, error) {
 	return []byte(fmt.Sprintf("%02d:%02d:%02d.%03d", h, m, s, ms)), nil
 }
 
-// UnmarshalText implements the encoding.TextUnmarshaler interface.
-func (dur *Duration) UnmarshalText(data []byte) (err error) {
-	parts := strings.SplitN(string(data), ":", 3)
+// DecodeDuration converts a timestamp into a Duration nanoseconds
+func DecodeDuration(data []byte) (*Duration, error) {
+	s := string(data)
+	if len(s) == 0 || strings.Compare(strings.ToLower(s), "undefined") == 0 {
```

Comment:
You don't need Compare here. `if len(s) == 0 || strings.ToLower(s) == "undefined" {`. This should also fix the build on 1.4